hardware_interface/src/resource_manager.cpp

```diff
@@ -884,6 +898,27 @@ class ResourceStorage
     }
   }
 
+  /**
+   * Returns the return type of the hardware component group state, if the return type is other
+   * than OK, then updates the return type of the group to the respective one
+   */
+  return_type update_hardware_component_group_state(
+    const std::string & group_name, const return_type & value)
+  {
+    // This is for the components that has no configured group
+    if (group_name.empty())
+    {
+      return value;
+    }
+    // If it is anything other than OK, change the return type of the hardware group state
+    // to the respective return type
+    if (value > hw_group_state_.at(group_name))
```

Comment:
```suggestion
    if (value != return_type::OK)
```
the docs clearly say what you wanted to write in code here ;) 
>  // If it is anything other than OK